### PR TITLE
refactor: panic on feature flag access without ConfigResolver [IDE-1974]

### PR DESCRIPTION
**Pull request by rrama**
### Description

Remove silent false/no-op when FolderConfig has no ConfigResolver in GetFeatureFlag and SetFeatureFlag.
Wire ConfigResolver in tests that were constructing incomplete FolderConfigs and not exercising flag reads.

Safe in prod code due to everything going through `wireConfigResolver`.
I did not do the rest of the functions in the file, this was enough for now.

FYI, this change is unrelated to the ticket, just the silent fail caused me some issues when I was adding tests for #1236.

### Checklist

- [x] Tests added and all succeed
- [ ] Regenerated mocks, etc. (`make generate`)
 - N/A
- [x] Linted (`make lint-fix`)
- [ ] README.md updated, if user-facing
 - N/A
- [ ] License file updated, if new 3rd-party dependency is introduced
 - N/A

### CODE REVIEW

File: internal/types/folder_config.go [hunk 1]
```diff
@@ -144,12 +140,8 @@ func (fc *FolderConfig) SetFeatureFlag(flag string, value bool) {
 	if fc == nil {
```
Comment (bastiandoetsch) [inline internal/types/folder_config.go:140]: [Should Fix] `SetFeatureFlag` was changed symmetrically with `GetFeatureFlag` to panic on nil `ConfigResolver`, but the test suite only adds a panic assertion for `GetFeatureFlag`. `SetFeatureFlag` has no corresponding `panics when ConfigResolver is nil` test case.

Fix: add a parallel test subcase in `TestFolderConfig_SetFeatureFlag` (or alongside the existing nil-receiver test):

```go
t.Run("panics when ConfigResolver is nil", func(t \*testing.T) {
    fc := &FolderConfig{}
    assert.Panics(t, func() { fc.SetFeatureFlag("myFlag", true) })
})
```

— AI review

**Comment by bastiandoetsch** [comment]
> ℹ️ This finding is in files not changed by this PR. It is non-blocking for this review — shared for awareness.

[Should Fix · non-blocking · ⚠ CONTRADICTED — Agents A and D verified safety] Several test files in `domain/snyk/scanner/` and `domain/snyk/scanner/scanner_precedence_test.go` construct `FolderConfig{FolderPath: ...}` without a `ConfigResolver` field. With the nil guard removed, any code path that calls `GetFeatureFlag`/`SetFeatureFlag` on these instances will now panic rather than return false.

Agents A and D independently verified that the current scan logic does NOT call `GetFeatureFlag`/`SetFeatureFlag` on these instances — so no immediate panic risk. However, if any future code in the scan path calls these methods, the tests will fail loudly with a panic rather than a clear error message.

Fix: Either document that these test instances intentionally never access feature flags, or proactively add `ConfigResolver: testutil.DefaultConfigResolver(engine)` to the relevant `FolderConfig` literals in scanner tests to future-proof them.

— AI review

**Comment by bastiandoetsch** [comment]
> ℹ️ This finding is in a file changed by this PR but at a line outside the diff. It is non-blocking for this review — shared for awareness.

[Suggestion · non-blocking] `FolderConfig.Conf()` is a public method (58 call sites) that now has an implicit panic precondition: if `ConfigResolver` is nil, calling `Conf().GetBool(...)` panics. The method's docstring does not warn callers of this precondition.

Two reviewers (semantic-analysis and adversarial) independently flagged this.

Fix: Add a docstring to `Conf()` at `internal/types/folder_config.go`:
```go
// Conf returns the configuration for prefix key access.
// Panics if ConfigResolver is nil — ensure ConfigResolver is initialized before calling.
func (fc \*FolderConfig) Conf() configuration.Configuration {
```

— AI review

File: internal/types/folder_config.go (hunk 1 above)
Comment (rrama) [reply internal/types/folder_config.go:140]: Just the one test is more than enough. Tbh I wouldn't have even done one if it wasn't already there. If someone changes it in the future we should catch it in review as silently failing is not a good thing.

**Comment by rrama** [comment]
> Fix: Either document that these test instances intentionally never access feature flags, or proactively add ConfigResolver: testutil.DefaultConfigResolver(engine) to the relevant FolderConfig literals in scanner tests to future-proof them.

No, if someone makes them call FFs the test will fail and they will catch the issue, maybe it will reveal the test is taking a different path they didn't want.


> Fix: Add a docstring to Conf() at internal/types/folder_config.go:
>
> ```go
> // Conf returns the configuration for prefix key access.
> // Panics if ConfigResolver is nil — ensure ConfigResolver is initialized before calling.
> func (fc \*FolderConfig) Conf() configuration.Configuration {
> ```

Sure fine, the old docstring was misleading. Done.

**Review by bastiandoetsch — APPROVED** [review]
